### PR TITLE
contrib/hx: new package (1.0.14)

### PR DESCRIPTION
**Pull request by ttyyls**

### CODE REVIEW

File: contrib/hx/template.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+pkgname = "hx"
+pkgver = "1.0.14"
+pkgrel = 0
+build_style = "makefile"
+make_cmd = "gmake"
+make_build_args = [
+    f"hx_git_hash=chimera-{pkgver}-r{pkgrel}",
+    "hx_version=1.0.14",
```
Comment (mhmdanas) [inline contrib/hx/template.py:8]:
```suggestion
    f"hx_version={pkgver}",
```

File: contrib/hx/template.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+sha256 = "7b7c494df149535f768b3aa449159928aec92b468a17f9eedebcea3dd21343d1"
+hardening = ["vis", "cfi"]
+# no tests defined
+options = ["!check"]
+
+
+def do_install(self):
+    self.install_bin("hx", name="hxed")
+    self.install_man("hx.1")
+    self.install_license("LICENSE")
```
[the remark this answers is not in the document]
Comment (ttyyls) [reply contrib/hx/template.py:25]: > Why not use the `install` make target?

name conflict and dubiouscompression of the manpage

[the remark this answers is not in the document]
Comment (mhmdanas) [reply contrib/hx/template.py:25]: You can patch it though.

[the remark this answers is not in the document]
Comment (ttyyls) [reply contrib/hx/template.py:25]: sure but why add 4 lines diff and clutter things up, after all it's just an executable and a manpage.

**Comment by nekopsykose** [comment]
in alpine it was renamed to hx-editor for instance

**Comment by ttyyls** [comment]
> this conflicts in name with helix for /usr/bin. not sure what to do about that tho

I handle it in do_install by installing the binary as hxed. hx-editor seemed a bit too clunky

**Comment by nekopsykose** [comment]
you know, i can't read sometimes

**Comment by nekopsykose** [comment]
i would also rename the manpage though then, so it matches (good) and just in case helix adds one later

**Comment by nekopsykose** [comment]
you need a patch for that

**Comment by ttyyls** [comment]
> you need a patch for that

fine :P

File: contrib/hx/template.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+pkgname = "hx"
+pkgver = "1.0.14"
+pkgrel = 0
+build_style = "makefile"
+make_cmd = "gmake"
+make_build_args = [
+    f"hx_git_hash=chimera-{pkgver}-r{pkgrel}",
```
Comment (q66) [inline contrib/hx/template.py:7]: pkgrel does not belong in stuff like that

File: contrib/hx/template.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    f"hx_version={pkgver}",
+]
+hostmakedepends = ["gmake"]
+pkgdesc = "Terminal Hex Editor"
+maintainer = "ttyyls <contact@behri.org>"
+license = "MIT"
+url = "https://github.com/krpors/hx"
+source = f"https://github.com/krpors/hx/archive/refs/tags/v{pkgver}.tar.gz"
```
Comment (q66) [inline contrib/hx/template.py:15]: {url}

File: contrib/hx/template.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+sha256 = "7b7c494df149535f768b3aa449159928aec92b468a17f9eedebcea3dd21343d1"
+hardening = ["vis", "cfi"]
+# no tests defined
+options = ["!check"]
+
+
+def do_install(self):
```
Comment (q66) [inline contrib/hx/template.py:22]: use install target

**Comment by q66** [comment]
cflags should be passed for linking and are not

**Comment by q66** [comment]
according to CI build log they are not

File: contrib/hx/template.py (hunk 5 above)
Comment (nekopsykose) [reply contrib/hx/template.py:22]: the install target would still need to manually rename the files after (or patch the makefile), and also uncompress the precompressed manpage. if anything it's more effort than installing 3 files here

Comment (q66) [reply contrib/hx/template.py:22]: the makefile needs patching regardless to fix other things so

**Comment by nekopsykose** [comment]
> according to CI build log they are not  

this seems to be a gmake default rule limitation, since we pass `LDFLAGS=ldflags CFLAGS=cflags` to gmake, and the project is using the default `hx: $(objects)` rule only. gmake itself seems to only pass ldflags for links..

should still be fixed, but gmake itself also has a bug here for that it seems

File: contrib/hx/template.py (hunk 5 above)
Comment (q66) [reply contrib/hx/template.py:22]: no incompressing is necessary, you just install one instead of the other

**Comment by q66** [comment]
the implicit rules should never be used

**Comment by ttyyls** [comment]
install target was a bit of a pain but I think the current patch addresses all of the other raised issues.

**Pull request merged**